packages/chain-mon/src/initialized-upgraded-mon/service.ts

```diff
@@ -0,0 +1,230 @@
+    name: 'sepolia',
+    l1StartingBlockTag: sepoliaConfig.l1StartingBlockTag,
+  },
+  11155420: {
+    name: 'op-sepolia',
+    l1StartingBlockTag: null,
+  },
+  5: {
+    name: 'goerli',
+    l1StartingBlockTag: goerliConfig.l1StartingBlockTag,
+  },
+  420: {
+    name: 'op-goerli',
+    l1StartingBlockTag: null,
+  },
+}
+
+const topic_initialized =
+  '0x7f26b83ff96e1f2b6a682f133852f6798a09c465da95921460cefb3847402498'
+
+const topic_upgraded =
+  '0xbc7cd75a20ee27fd9adebab32041f755214dbc6bffa90cc0225b39da2e5c2d3b'
+
+type InitializedUpgradedMonOptions = {
+  rpc: Provider
+  startBlockNumber: number
+  contracts: string
+}
+
+type InitializedUpgradedMonMetrics = {
+  initializedCalls: Counter
+  upgradedCalls: Counter
+  unexpectedRpcErrors: Counter
+}
+
+type InitializedUpgradedMonState = {
+  chainId: number
+  highestUncheckedBlockNumber: number
+  contracts: Array<{ label: string; address: string }>
+}
+
+export class InitializedUpgradedMonService extends BaseServiceV2<
+  InitializedUpgradedMonOptions,
+  InitializedUpgradedMonMetrics,
+  InitializedUpgradedMonState
+> {
+  constructor(
+    options?: Partial<InitializedUpgradedMonOptions & StandardOptions>
+  ) {
+    super({
+      version,
+      name: 'initialized-upgraded-mon',
+      loop: true,
+      options: {
+        loopIntervalMs: 1000,
+        ...options,
+      },
+      optionsSpec: {
+        rpc: {
+          validator: validators.provider,
+          desc: 'Provider for network to monitor balances on',
+        },
+        startBlockNumber: {
+          validator: validators.num,
+          default: -1,
+          desc: 'L1 block number to start checking from',
+          public: true,
+        },
+        contracts: {
+          validator: validators.str,
+          desc: 'JSON array of [{ label, address }] to monitor contracts for',
+          public: true,
+        },
+      },
+      metricsSpec: {
+        initializedCalls: {
+          type: Gauge,
+          desc: 'Successful transactions to tracked contracts emitting initialized event',
+          labels: ['label', 'address'],
+        },
+        upgradedCalls: {
+          type: Gauge,
+          desc: 'Successful transactions to tracked contracts emitting upgraded event',
+          labels: ['label', 'address'],
+        },
+        unexpectedRpcErrors: {
+          type: Counter,
+          desc: 'Number of unexpected RPC errors',
+          labels: ['section', 'name'],
+        },
+      },
+    })
+  }
+
+  protected async init(): Promise<void> {
+    // Connect to L1.
+    await waitForProvider(this.options.rpc, {
+      logger: this.logger,
+      name: 'L1',
+    })
+
+    this.state.chainId = await getChainId(this.options.rpc)
+
+    const l1StartingBlockTag = networks[this.state.chainId].l1StartingBlockTag
+
+    if (this.options.startBlockNumber === -1) {
+      const block_number =
+        l1StartingBlockTag != null
+          ? (await this.options.rpc.getBlock(l1StartingBlockTag)).number
```

Comment:
```suggestion
          ? (await this.options.rpc.getBlockNumber()
```
this is a nit feel freee to ignore. seems like a getBlockNumber exists

Reply:
ignoring this because we want to track since the first bedrock block if `this.options.startBlockNumber === -1`